examples/ex71.hpp

```diff
@@ -0,0 +1,587 @@
+// shared implementation ex71p/ex71 for the AD integrands and
+// the handconded integrators
+
+
+#ifndef EXAMPLE71_H
+#define EXAMPLE71_H
+
+#include "mfem.hpp"
+#include <memory>
+#include <iostream>
+#include <fstream>
+
+
+namespace mfem
+{
+
+
+template<typename DType, typename MVType>
+class MyQFunctorJ
+{
+public:
+   DType operator()(const mfem::Vector& vparam, MVType& uu)
+   {
+      double pp=vparam[0];
+      double ee=vparam[1];
+      double ff=vparam[2];
+
+      DType  u=uu[3];
+      DType  norm2=uu[0]*uu[0]+uu[1]*uu[1]+uu[2]*uu[2];
+
+      DType rez= pow(ee*ee+norm2,pp/2.0)/pp-ff*u;
+      return rez;
+   }
+
+   void operator()(const mfem::Vector& vparam, MVType& uu, MVType& rr)
+   {
+      double pp=vparam[0];
+      double ee=vparam[1];
+      double ff=vparam[2];
+
+      DType norm2=uu[0]*uu[0]+uu[1]*uu[1]+uu[2]*uu[2];
+      DType tvar=pow(ee*ee+norm2,(pp-2.0)/2.0);
+
+      rr[0]=tvar*uu[0];
+      rr[1]=tvar*uu[1];
+      rr[2]=tvar*uu[2];
+      rr[3]=-ff;
+
+   }
+};
+
+
+typedef ADQFunctionTJ<MyQFunctorJ,4> pLapIntegrandTJ;
+
+
+template<typename DType, typename MVType>
+class MyQFunctorH
+{
+public:
+   DType operator()(const mfem::Vector& vparam, MVType& uu)
+   {
+      double pp=vparam[0];
```

Comment:
I think an `enum class` would be better than a special index here.

Reply:
We do not know in advance the number of the variables - this means that we have to define enum for every separate case.